tools/Microsoft.TemplateSearch.TemplateDiscovery/PackChecking/PackSourceChecker.cs

```diff
@@ -156,10 +156,10 @@ await foreach (ITemplatePackageInfo sourceInfo in packProvider.GetCandidatePacks
                 var packageInfo = await provider.GetPackageInfoAsync(package.Name, cancellationToken).ConfigureAwait(false);
                 if (packageInfo == default)
                 {
-                    Console.WriteLine($"Package {package.Name} cannot be verified.");
-                    throw new Exception($"Package {package.Name} is missing and cannot be verified.");
+                    Console.WriteLine($"Package {package.Name} no longer exists in the provider.");
+                    scanningStats.RemovedTemplatePacks.Add(package);
                 }
-                if (packageInfo.Removed)
+                else if (packageInfo.Removed)
```

Comment:
Finally, the core change. Templates that are unlisted are added to the 'filter me out from future use' cache, and it stands to reason that removed one could be too. This is a per-package-version filter so newer versions that are added would be re-added to the cache.

Reply:
What are the different cases where packageInfo will be default versus removed?

Reply:
`Removed` in this tool's vocabulary maps to 'unlisted' in NuGet terminology. `default` is because the `PackageInfo` type is a struct and if not found the `GetPackageInfoAsync` method will return the `default` struct - essentially mocking out a nullable type here.